branchpro/tests/test_models.py

```diff
@@ -20,3 +21,35 @@ def test_simulate(self):
         forward_model = bp.ForwardModel()
         with self.assertRaises(NotImplementedError):
             forward_model.simulate(0, 1)
+
+
+class TestBrachProModelClass(unittest.TestCase):
+    """
+    Test the 'BranchProModel' class.
+    """
+    def test__init__(self):
+        branch_model = bp.BranchProModel(0, [1])
+        self.assertEqual(branch_model._serial_interval, np.array([1]))
+        self.assertEqual(branch_model._initial_r, 0)
+        self.assertEqual(branch_model._normalizing_const, 1)
+
+        with self.assertRaises(ValueError):
+            bp.BranchProModel(0, [0])
+
+        with self.assertRaises(TypeError):
+            bp.BranchProModel('0', [1])
+            bp.BranchProModel(0, 1)
+
+    def test_simulate(self):
+        branch_model_1 = bp.BranchProModel(2, [1, 2, 3, 2, 1])
+        simulated_sample_model_1 = branch_model_1.simulate(1, [2, 4])
+        new_simulated_sample_model_1 = branch_model_1.simulate(1, [0, 2, 4])
+        self.assertEqual(len(simulated_sample_model_1), 2)
```

Comment:
Could you test the shape here instead of the length? This just makes sure that the simulation results are actually 1dimensional, i.e.
```
self.assertEqual(simulated_sample_model_1.shape, (2, ))
```